Katana/Store.swift

```diff
@@ -191,18 +191,31 @@ open class Store<S: State, D: SideEffectDependencyContainer>: PartialStore<S> {
   /// Whether the store is ready to execute operations
   public private(set) var isReady: Bool
   
-  /**
-   The dependencies used in the side effects
-   
-   - seeAlso: `SideEffect`
-   */
+  /// The dependencies used in the side effects
+  ///
+  ///- seeAlso: `SideEffect`
   public var dependencies: D!
   
+  /// The dependencies used to initialize katana
+  public struct Configuration {
+    /// On this queue katana runs all the listeners
+    let mainAsyncProvider: AsyncProvider
+    
+    public init(
+      mainAsyncProvider: AsyncProvider = DispatchQueue.main
+    ) {
+      self.mainAsyncProvider = mainAsyncProvider
+    }
+  }
+  
   /// The context passed to the side effect
   private var sideEffectContext: SideEffectContext<S, D>!
   
+  /// The queue used to handle the listeners
+  fileprivate var mainAsyncProvider: AsyncProvider!
+    
   /// The queue used to handle the `StateUpdater` items
-  lazy fileprivate var stateUpdaterQueue: DispatchQueue = {
+  fileprivate var stateUpdaterQueue: DispatchQueue! = {
     let d = DispatchQueue(label: "katana.stateupdater", qos: .userInteractive)
```

Comment:
why these are banged?

Reply:
so that I can release them (setting them to nil) when trying to garbage collect the store. I think they get captured in some closure for the interceptor chain, but `mi hai messo il tarlo` and I'll investigate more

Reply:
removed the bangs, as they are not really captured